src/main/java/com/emc/pravega/perf/PravegaReaderWorker.java

```diff
@@ -30,60 +28,44 @@
+public class PravegaReaderWorker extends ReaderWorker  {
     public final static AtomicInteger eventCount = new AtomicInteger(0);
-    public final long totalEvents;
+
     private final EventStreamReader<String> reader;
-    private final int secondsToRun;
-    private final Instant StartTime;
-    private final int timeout;
-    private final PerfStats stats;
     private final String readerId;
 
     PravegaReaderWorker(int readerId, int secondsToRun, Instant start,
-                        ClientFactory factory, PerfStats stats,
-                        String readergrp, long totalEvents, int timeout) {
-        this.readerId = Integer.toString(readerId);
-        this.secondsToRun = secondsToRun;
-        this.StartTime = start;
-        this.stats = stats;
-        this.totalEvents = totalEvents;
-        this.timeout = timeout;
+                        PerfStats stats, String readergrp, long totalEvents,
+                        int timeout, ClientFactory factory) {
+        super(readerId, secondsToRun, start, stats, readergrp, totalEvents, timeout);
 
+        this.readerId = Integer.toString(readerId);
         reader = factory.createReader(
                 this.readerId, readergrp, new UTF8StringSerializer(), ReaderConfig.builder().build());
     }
 
-    public void cleanupEvents(PerfStats drainStats) throws ReinitializationRequiredException {
-        EventRead<String> event;
-        String ret = null;
-        do {
-            final Instant startTime = Instant.now();
-            event = reader.readNextEvent(timeout);
-            ret = event.getEvent();
-            if (ret != null) {
-                drainStats.recordTime(null, startTime, ret.length());
-            }
-        } while (ret != null);
-        drainStats.printTotal(Instant.now());
-    }
-
     @Override
-    public Void call() throws ReinitializationRequiredException, ExecutionException {
-        EventRead<String> event = null;
-        String ret = null;
+    public  String readData() throws Exception {
         try {
-            do {
-                event = reader.readNextEvent(timeout);
-                ret = event.getEvent();
-                if (ret != null) {
-                    stats.recordTime(null, Instant.ofEpochMilli(Long.parseLong(ret.split(",")[0])), ret.length());
-                }
-            } while ((Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun) &&
-                    (eventCount.incrementAndGet() < totalEvents));
-        } finally {
-            reader.close();
+            return reader.readNextEvent(timeout).getEvent();
+        } catch(ReinitializationRequiredException e){
+            throw new Exception("ReinitializationRequiredException exception while reading data ");
         }
-        return null;
     }
+
+    @Override
+    public void close(){
+        reader.close();
+    }
+
+    @Override
+    public long eventCountIncrementAndGet() {
```

Comment:
This should not be public

Reply:
no, this is an abstract methods must be overridden in derived class , need to be public.

Reply:
removed these methods